components/src/core/Drawer/DrawerNavGroup.tsx

```diff
@@ -112,31 +243,153 @@ function NavigationListItem(item: NavItem, props: DrawerNavGroupProps): ReactNod
+    }
+
+    function getActionComponent(): JSX.Element {
+        if (chevron) {
+            return null;
+        }
+        if (!subItems) {
+            return null;
+        }
+        if (collapseIcon) {
+            return (
+                <div
+                    onClick={(e): void => {
+                        if (e) {
+                            expandHandler();
+                        }
+                    }}
+                    className={classes.expandIcon}
+                >
+                    {expanded ? collapseIcon : getExpandIcon()}
+                </div>
+            );
+        }
+
+        return (
+            <div
+                onClick={(e): void => {
+                    if (e) {
+                        expandHandler();
+                    }
+                }}
+                className={`${classes.expandIcon} ${expanded ? 'expanded' : ''}`}
+            >
+                {getExpandIcon()}
+            </div>
+        );
+    }
+
+    const actionComponent = getActionComponent();
+
+    // 2 indents for top level nav items
+    // 2, 4, 6, ... for secondary level and beyond
+    const paddingLeft = theme.spacing(depth ? (depth - 1) * 4 + 2 : 2);
+
+    const active = activeItem === itemID;
+
+    const statusColor = (navItem as NavItem).statusColor;
+
     return (
         <div style={{ position: 'relative' }} className={`${classes.listItem} ${active && classes.listItemNoHover}`}>
-            {active && <div className={classes.active} style={{ backgroundColor: activeBackgroundColor }} />}
+            {active && (
+                <div
+                    className={`${classes.active} ${activeBackgroundShape === 'rectangular' ? 'rectangular' : ''}`}
+                    style={{ backgroundColor: activeBackgroundColor }}
+                />
+            )}
             <InfoListItem
                 dense
                 title={title}
                 subtitle={subtitle}
-                divider={
-                    item.divider === undefined ? (divider ? 'full' : undefined) : item.divider ? 'full' : undefined
-                }
+                divider={divider ? 'full' : undefined}
                 statusColor={statusColor}
                 fontColor={active ? activeFontColor : fontColor}
                 icon={icon}
                 iconColor={active ? activeIconColor : iconColor}
                 chevron={chevron}
+                rightComponent={
+                    (actionComponent || rightComponent) && (
+                        <div style={{ display: 'flex', alignItems: 'center' }}>
+                            {rightComponent}
+                            {actionComponent}
+                        </div>
+                    )
+                }
                 backgroundColor={'transparent'}
-                onClick={(): void => action()}
+                onClick={(): void => {
+                    action();
+                    expandHandler();
+                }}
```

Comment:
So we always expand/collapse on click of the row, even if they have a onClick function defined for it? I'd be inclined to not call expandHandler if props.onClick is defined